File: 2257/solution.java
class Solution {
    public int countUnguarded(int m, int n, int[][] guards, int[][] walls) {
        char[][] grid = new char[m][n];
        int result = 0;
        for(int[] guard:guards)
            grid[guard[0]][guard[1]] = 'g';
        
        for(int[] wall:walls)
            grid[wall[0]][wall[1]] = 'w';

        int r,c,cr,cc;
        for(int[] guard:guards){
            r = guard[0];
            c = guard[1];
            //left
            cc = c-1;
            while(cc>=0 && grid[r][cc]!='g' && grid[r][cc]!='w')
                grid[r][cc--] = 'o';
            //right
            cc = c+1;
            while(cc<n && grid[r][cc]!='g' && grid[r][cc]!='w')
                grid[r][cc++] = 'o';
            //top
            cr = r-1;
            while(cr>=0 && grid[cr][c]!='g' && grid[cr][c]!='w')
                grid[cr--][c] = 'o';
            //bottom
            cr = r+1;
            while(cr<m && grid[cr][c]!='g' && grid[cr][c]!='w')
                grid[cr++][c] = 'o';
            
        }

        for(char[] chr:grid){
            for(char ch:chr){
                if(ch=='\u0000')
                    result++;
            }
        }
        return result;

    }
}

File: 2257/solution.py
class Solution:
    def countUnguarded(self, m: int, n: int, guards: List[List[int]], walls: List[List[int]]) -> int:
        grid = [['']*n for _ in range(m)]
        result = 0
        for r,c in guards:
            grid[r][c] = 'g'
        for r,c in walls:
            grid[r][c] = 'w'
        
        for r,c in guards:
            cc = c-1
            #left
            while cc>=0 and grid[r][cc] not in ['g','w']:
                grid[r][cc] = 'o'
                cc-=1
            cc = c+1
            #right
            while cc<n and grid[r][cc] not in ['g','w']:
                grid[r][cc] = 'o'
                cc+=1
            cr = r-1
            #top
            while cr>=0 and grid[cr][c] not in ['g','w']:
                grid[cr][c] = 'o' 
                cr-=1
            cr = r+1
            #bottom
            while cr<m and grid[cr][c] not in ['g','w']:
                grid[cr][c] = 'o'
                cr+=1
        
        for cnt in grid:
            result+=cnt.count('')
        return result
